### PR TITLE
Send full stack trace to remote endpoint on internal error

### LINKED ISSUE #119: Provide exception details with ResponseError
Currently, we do not provide any information about the exception that caused a ResponseError.
In RemoteProxy we should `setData` to set the string representation of the throwable.

### PR DESCRIPTION
**Pull request by spoenemann**
Fixes #119.

### CODE REVIEW

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java [hunk 1]
```diff
@@ -47,8 +49,13 @@
 		} else {
 			LOG.log(Level.SEVERE, "Internal error: " + throwable.getMessage(), throwable);
 			ResponseError error = new ResponseError();
-			error.setMessage("Internal error, please look at the server's logs.");
+			error.setMessage("Internal error.");
```
Comment (akosyakov) [inline org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java]: probably `throwable.getMessage()` should be also part of a message

**Comment by akosyakov** [comment]
I wonder whether it is always the good thing to expose an internal exception information to the client. In VS code context it should not harm, not sure about servers running remotely.

File: org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java (hunk 1 above)
Comment (spoenemann) [reply org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/RemoteEndpoint.java]: The first line of the stack trace includes the message.

**Comment by spoenemann** [comment]
In case you want to reduce the amount of information sent to the client, you can create the RemoteEndpoint with a different exception handler. My change only affects the default exception handler.

**Comment by spoenemann** [comment]
I'll merge this so I can prepare release 0.3.0

**Pull request merged**